cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterServiceMain.java

```diff
@@ -290,44 +290,64 @@ private InetAddress getLocalHost() {
   /**
    * Returns a map from system service name to a map from property to configuration key.
    */
-  private Map<String, Map<String, String>> getConfigKeys() {
-    Map<String, Map<String, String>> configKeys = Maps.newHashMap();
-
-    configKeys.put(Constants.Service.LOGSAVER,
-                   ImmutableMap.of("default", Constants.LogSaver.NUM_INSTANCES,
-                                   "max", Constants.LogSaver.MAX_INSTANCES));
-    configKeys.put(Constants.Service.TRANSACTION,
-                   ImmutableMap.of("default", Constants.Transaction.Container.NUM_INSTANCES,
-                                   "max", Constants.Transaction.Container.MAX_INSTANCES));
-    configKeys.put(Constants.Service.METRICS_PROCESSOR,
-                   ImmutableMap.of("default", Constants.MetricsProcessor.NUM_INSTANCES,
-                                   "max", Constants.MetricsProcessor.MAX_INSTANCES));
-    configKeys.put(Constants.Service.METRICS,
-                   ImmutableMap.of("default", Constants.Metrics.NUM_INSTANCES,
-                                   "max", Constants.Metrics.MAX_INSTANCES));
-    configKeys.put(Constants.Service.STREAMS,
-                   ImmutableMap.of("default", Constants.Stream.CONTAINER_INSTANCES,
-                                   "max", Constants.Stream.MAX_INSTANCES));
-    configKeys.put(Constants.Service.DATASET_EXECUTOR,
-                   ImmutableMap.of("default", Constants.Dataset.Executor.CONTAINER_INSTANCES,
-                                   "max", Constants.Dataset.Executor.MAX_INSTANCES));
-    configKeys.put(Constants.Service.EXPLORE_HTTP_USER_SERVICE,
-                   ImmutableMap.of("default", Constants.Explore.CONTAINER_INSTANCES,
-                                   "max", Constants.Explore.MAX_INSTANCES));
-    return configKeys;
+  private Set<ServiceResourceKeys> getServiceResources() {
+    Set<ServiceResourceKeys> serviceResourceKeys = Sets.newHashSet();
+
+    serviceResourceKeys.add(new ServiceResourceKeys(cConf,
```

Comment:
this seems like the same set constructed as in AbstractMasterCheck.
No need to construct it twice, separately. Two copies can also lead to inconsistency, if a change is made to one place and not the other.